code/modules/urist/items/saintsandsins.dm

```diff
@@ -0,0 +1,681 @@
+	Key spells may only be used on your turn. Reaction spells may be used in reaction to another player's spell, even if it is not targeting you. You may cast up to 2 spells on your turn, and draw 1 card after ending your turn. You may only use one reaction spell per round. <BR><BR>\
+	Damage types: <BR><BR>\
+	Fire | Water | Earth | Wind | Decay | Holy <BR><BR>\
+	Suitable for 2-4 players."
+
+/obj/item/pack/saintsandsins
+	name = "\improper Saints and Sins builder pack"
+	desc = "A builder pack for hit collectable card game, Saints and Sins. This one will help any aspiring deck builder get started."
+	icon = 'icons/urist/items/saintsandsins.dmi'
+	icon_state = "card_pack_saintsandsins"
+	var/pack_type = CARDPACK_BUILDER
+	var/how_many_cards = 21 /// Boosters give less cards than the builder packs.
+
+
+/obj/item/pack/saintsandsins/Initialize()
+	. = ..()
+	var/list/heretic_weights = build_card_weights_list(pack_type, /singleton/hierarchy/sas_card/heretic)
+	var/list/spell_weights = build_card_weights_list(pack_type, /singleton/hierarchy/sas_card/spell)
+	for(var/i in 1 to how_many_cards)
+		var/datum/playingcard/P = new
+		var/singleton/hierarchy/sas_card/picked_card
+		if(i == 1) // Just pick one Heretic.
+			picked_card = pickweight(heretic_weights)
+			P.card_icon = "card_cardemon"
+		else
+			picked_card = pickweight(spell_weights)
+			P.card_icon = "card_cardemon_Silver"
+		P.name = picked_card.name
+		P.desc = picked_card.desc
+		P.back_icon = "card_back_cardemon"
+		cards += P
+
+/obj/item/pack/saintsandsins/proc/build_card_weights_list(pack_type, root_type)
+	. = list()
+	var/list/card_types = GET_SINGLETON_SUBTYPE_MAP(root_type)
+	for(var/card_type in card_types)
+		var/singleton/hierarchy/sas_card/C = card_types[card_type]
+		if(pack_type in C.pack_probability)
+			.[C] = C.pack_probability[pack_type] || 1
+
+/**
+ * Booster packs
+ * Can have special cards not found in builder pack.
+ * Add new boosters to /obj/random/saintsandsins_packs please.
+ */
+
+/obj/item/pack/saintsandsins/booster
+	name = "\improper Saints and Sins booster pack"
+	var/booster_type = "Basic Witch Kit"
+	pack_type = CARDPACK_BASICWITCH
+	how_many_cards = 11
+
+/obj/item/pack/saintsandsins/booster/Initialize()
+	. = ..()
+	desc = "A booster pack for hit collectable card game, Saints and Sins. This one is labeled '[booster_type]'."
+
+/obj/item/pack/saintsandsins/booster/alchemist
+	booster_type = "Wicked Alchemist Kit"
+	pack_type = CARDPACK_ALCHEMIST
+
+/obj/item/pack/saintsandsins/booster/fatebringer
+	booster_type = "Fatebringer Kit"
+	pack_type = CARDPACK_FATEBRINGER
+
+/*
+// Builder pack cards can appear in other packs to fluff them up. Pack-specific are less likely to do so, if ever.
+*/
+/singleton/hierarchy/sas_card
+	var/desc
+	var/ability
+	var/pack_probability
```

Comment:
Is there any reason this is a Hierarchy type? AFAICT it doesn't use the Hierarchy API anywhere.

At first glance, it looks like it was PLANNED to be a full-on minigame, but was turned into just 'read the desc and do what it says manually'.

Reply:
Gave datums a shot and it runtimed+broke, can't think of a better way.